test/test_um2netcdf.py

```diff
@@ -651,6 +669,355 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
         assert ua_plev_cube.units == orig  # nothing should happen as there's no cube.units
 
 
+def to_iris_DimCoord(points_and_name_func):
+
+    def DimCoord_maker():
+        points, name = points_and_name_func()
+        return iris.coords.DimCoord(
+            points=points,
+            standard_name=name
+        )
+
+    return DimCoord_maker
+
+
+@pytest.fixture
+@to_iris_DimCoord
+def lat_river_coord():
+    # iris DimCoord imitating UM V7.3s 1x1 degree river grid.
+    # Must have length 180.
+    lat_river_points = np.arange(-90., 90, dtype="float32") + 0.5
+    return lat_river_points, um2nc.LATITUDE
```

Comment:
Wow, the decorator approach is less explicit but really neatens the fixtures. Nice!